src/Map.php

```diff
@@ -963,7 +976,8 @@ protected function walk_backward(callable $code)
     {
         for (end($this->array); null !== ($hash = key($this->array)); prev($this->array)) {
             $key   = $this->hash_to_key($hash);
-            $value =& current($this->array);
+            $current = current($this->array);
+            $value =& $current;
```

Comment:
Why the intermediate variable?

Reply:
PHPUnit was failing until I added it. Here the error:

```
PHPUnit 5.7.22 by Sebastian Bergmann and contributors.

...............................................................  63 / 121 ( 52%)
.........................E................................      121 / 121 (100%)

Time: 145 ms, Memory: 4.00MB

There was 1 error:

1) Haldayne\Boost\MapTest::test_last
Only variables should be assigned by reference

/home/deevus/Projects/forks/boost/src/Map.php:979
/home/deevus/Projects/forks/boost/src/Map.php:936
/home/deevus/Projects/forks/boost/src/Map.php:193
/home/deevus/Projects/forks/boost/tests/MapTest.php:197

ERRORS!
Tests: 121, Assertions: 220, Errors: 1.
```